great_expectations/datasource/generator/s3_generator.py

```diff
@@ -94,10 +91,7 @@ def __init__(self,
             boto3_options = {}
         self._max_keys = max_keys
         self._iterators = {}
-        try:
-            self._s3 = boto3.client('s3', **boto3_options)
-        except TypeError:
```

Comment:
It is also possible to use AttributeError here, but this module will need to use boto3 nevertheless so why not make it fail on the first place.